### ISSUE TITLE
Networkd does not properly resume interfaces after suspend

### ISSUE THREAD

**Issue by LaserEyess** [systemd/systemd#16724]
**systemd version the issue has been seen with**
> systemd 246 (246.1-1-arch)
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > networkd brings up the interfaces 

**Unexpected behaviour you saw**
 > broken network and networkctl lists all managed, routable interfaces as failed

**Steps to reproduce the problem**

`systemctl suspend` and then resume. So far I can reproduce it any time. Simply restarting `systemd-networkd.service` fixes this.

Log file with `Environment=SYSTEMD_LOG_LEVEL=debug`: https://0x0.st/iYMO.log

My network:
```
IDX LINK   TYPE      OPERATIONAL SETUP     
  1 lo     loopback  carrier     unmanaged 
  2 enp4s0 ether     enslaved    configured
  3 bond1  bond      routable    configured
  4 wgs0   wireguard routable    configured
```
Both `bond1` and `wgs0` are internet facing interfaces and they both return as failed after resume. I can't ping anything routed through either interface, although local addresses still seem to work e.g. `ping 10.0.1.1` (my router).

``` 
$ ls /etc/systemd/network
10-bond1.netdev  10-bond1.network  20-etho.network  50-wgs0.netdev  50-wgs0.network  90-dracut-eth.network
```

``` 
$ cat 10-bond1.netdev
[NetDev]
Name=bond1
Kind=bond

[Bond]
Mode=balance-xor 
```

```
$ cat 10-bond1.network
[Match]
Name=bond1

[Network]
Address=10.0.1.5/24
Gateway=10.0.1.1
DNS=10.0.1.1
```

```
$ cat 20-etho.network
[Match]
PermanentMACAddress=aa:bb:cc:dd:ee:ff

[Network]
Bond=bond1
```

```
$ sudo -E cat 50-wgs0.network
[Match]
Name = wgs0

[Network]
Address = 10.20.30.4/24

[Route]
Gateway = 10.20.30.1
Destination = 10.100.0.0/24

[Route]
Gateway = 10.20.30.1
Destination = 10.20.30.0/24
```

**Comment by LaserEyess** [comment]
Another thing I've discovered, this is only an issue for static IP address. Seems that my laptop, which gets its address from DHCP does not have this issue, however I have a single interface on my laptop with a static IP address managed with networkd and it fails.

**Comment by LaserEyess** [comment]
This exists on 246, but not 245.7. Never bisected systemd before, but if that's necessary I can try it.

**Comment by yuwata** [comment]
Duplicate of #16696, and fixed by #16697.
